web/src/recent_view_ui.ts

```diff
@@ -497,6 +497,90 @@ export function process_messages(
     }
 }
 
+function get_unread_count(data: ConversationData): number {
+    if (data.type === "private") {
+        return unread.num_unread_for_user_ids_string(data.to_user_ids);
+    }
+    return unread.num_unread_for_topic(data.stream_id, data.topic);
+}
+
+function get_display_reply_to(conversation_data: ConversationData): string {
+    assert(conversation_data.type === "private");
+    const user_ids = conversation_data.to_user_ids
+        .split(",")
+        .map((user_id) => Number.parseInt(user_id, 10));
+    return message_store.get_pm_full_names(user_ids);
+}
+
+function get_pm_url(conversation_data: ConversationData): string {
+    assert(conversation_data.type === "private");
+    const user_ids = conversation_data.to_user_ids
+        .split(",")
+        .map((user_id) => Number.parseInt(user_id, 10));
+    const pm_url = people.pm_with_url_from_user_ids(user_ids);
+    assert(pm_url !== undefined);
+    return pm_url;
+}
+
+// This mirrors the structure of `echo.build_display_recipient`, using data
+// found in `ConversationData`.
+function get_display_recipient(conversation_data: ConversationData): DisplayRecipientUser[] {
+    assert(conversation_data.type === "private");
+    const user_ids = conversation_data.to_user_ids
+        .split(",")
+        .map((user_id) => Number.parseInt(user_id, 10));
+    const to_users = user_ids.map((user_id) => people.get_by_user_id(user_id));
+
+    let sender_in_display_recipients = false;
+    const display_recipient: DisplayRecipientUser[] = to_users.map((user) => {
+        const email = user.email.trim();
+        if (user === undefined) {
+            // For unknown users, we return a skeleton object.
+            //
+            // This allows us to support zephyr mirroring situations
+            // where the server might dynamically create users in
+            // response to messages being sent to their email address.
+            // See `echo.build_display_recipient` for more on this.
+            return {
+                // TODO(evy): What to do here for typing purposes? Or can we assume all users are defined here?
+                // I don't know much aobut zephyr and when it's relevant.
```

Comment:
question here!

Reply:
For recent view, I think it makes sense to treat them as normal users unless we have problems generating their avatar.

Reply:
I'll let @timabbott comment before removing it then